dashboard/app/models/script_level.rb

```diff
@@ -445,10 +445,11 @@ def summarize(include_prev_next=true, for_edit: false)
     summary
   end
 
-  def summarize_for_lesson_show(can_view_teacher_markdown)
+  def summarize_for_lesson_show(current_user, can_view_teacher_markdown)
```

Comment:
Could you double-check that the result of this method (and other summarize methods that now take current_user) are not cached anywhere?  We had a bug awhile back where summarize results were being cached (for a cached level?) and that caused the page to be displayed in the wrong language to some users.

Reply:
Also what is the best way to find out if something like this is being cached?

Reply:
No, for the result to be cacheable, the result has to be the same for all users (and the method should therefore not need to take the current_user).  Here's the PR I was referring to, for some additional background: https://github.com/code-dot-org/code-dot-org/pull/40035.

The best way I can think of is to trace through where these methods are being called and make sure that results are only ever returned to the user and not saved in a ruby object.  Maybe others have better ideas?

Reply:
It looks like the `summarize_for_lesson_show` methods I am updating are just used by other `summarize_for_lesson_show` methods further up the chain. It looks like if you follow the chain up we do save the result of `summarize_for_lesson_show` on lesson here: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/lessons_controller.rb#L30.

Reply:
I think that instance might be ok.  It looks like we're storing it on the controller (and not a model object that's cached).  (I never totally understood the lifecycle of a rails controller but I believe the instance variables are somehow per-request?)

Reply:
> I think that instance might be ok. It looks like we're storing it on the controller (and not a model object that's cached). (I never totally understood the lifecycle of a rails controller but I believe the instance variables are somehow per-request?)

This sounds right to me. `@lesson_data` is an instance variable on the controller, which will only persist for the lifetime of the request, so the method call Dani linked to seems safe. on the other hand, class variables, starting with `@@`, are cached across requests. `@@unit_cache` is one example of this.